azurerm/internal/services/automation/automation_runbook_resource.go

```diff
@@ -298,20 +305,22 @@ func resourceArmAutomationRunbookDelete(d *schema.ResourceData, meta interface{}
 	return nil
 }
 
-func expandContentLink(d *schema.ResourceData) automation.ContentLink {
-	inputs := d.Get("publish_content_link").([]interface{})
+func expandContentLink(inputs []interface{}) *automation.ContentLink {
+	if len(inputs) == 0 {
```

Comment:
We should also check for nil 
```suggestion
	if len(inputs) == 0 || inputs[0] == nil {
```